src/k8s/pkg/k8s/setup/k8s_dqlite.go

```diff
@@ -52,9 +53,13 @@ func JoinK8sDqliteCluster(ctx context.Context, state *state.State, snap snap.Sna
 	return nil
 }
 
-func LeaveK8sDqliteCluster(ctx context.Context, snap snap.Snap, hostname string) error {
-	// TODO: Get dqlite port from config
-	address := fmt.Sprintf("%s:%d", hostname, dqlite.K8sDqliteDefaultPort)
+func LeaveK8sDqliteCluster(ctx context.Context, snap snap.Snap, state *state.State) error {
+	clusterConfig, err := utils.GetClusterConfig(ctx, state)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster config: %w", err)
+	}
+
+	address := fmt.Sprintf("%s:%d", state.Address().Hostname(), clusterConfig.K8sDqlite.Port)
```

Comment:
here, you would use the subnet configuration to find the node IP that matches.

Its fine if we leave the address as-is for now, and rely on the dqlite default to work well

Reply:
Alright, let's just leave the address out of the cluster config for now.